osu.Game/Online/API/APIAccess.cs

```diff
@@ -255,6 +253,19 @@ public RegistrationRequest.RegistrationRequestErrors CreateAccount(string email,
             return null;
         }
 
+        private void fetchFriends(Action onSuccess, Action onFail)
```

Comment:
these callbacks seem sort of overdone to be honest, i'd probably just inline.

Reply:
setting `State` to `Failing` inside this method? don't think that'll be a good idea given that the method is allowed to be called anywhere from the whole class.

could just revert this extraction and call it a day.

Reply:
arguing that a private method has to be safely callable from any point in its class does not make a lot of sense to me.

i'm just saying it's the only usage for now, no reason to overcomplicate matters. if someone doesn't look at what methods they're calling and if they can call them in the first place without breaking stuff, they probably shouldn't be programming in the first place.

Reply:
arguing that a private method has to be safely callable from any point in its class does not make a lot of sense to me.

i'm just saying it's the only usage for now, no reason to overcomplicate matters. if someone doesn't look at what methods they're calling and if they can call them in the first place without breaking stuff, they probably shouldn't be programming in the first place.

Reply:
I've moved this in locally and un-nested for better flow.